src/components/Template.js

```diff
@@ -137,4 +137,17 @@ function transformHelpersToHogan(helpers, compileOptions, data) {
   );
 }
 
-export default Template;
+// Resolve transformData before Template, so transformData is always called
+// even if the data is the same. Allowing you to dynamically inject conditions in
+// transformData that will force re-rendering
+const withTransformData =
+  TemplateToWrap =>
+    props => {
+      const data = props.data === undefined ? {} : props.data; // eslint-disable-line react/prop-types
```

Comment:
I think it happens in currentRefinedValues where you can easily `create` items via attributes and those might not yet be refined.